### ISSUE TITLE
false error on untyped `__init__`

### ISSUE THREAD

**Issue by KotlinIsland** [KotlinIsland/basedmypy#320]
```py
# mypy: allow-untyped-defs
class A:
    def __init__(self, foo): ...  # error: Function is missing a type annotation for one or more arguments
```

There should be no error here
